todo_app/app.py

```diff
@@ -8,4 +9,27 @@
 
 @app.route('/')
 def index():
-    return 'Hello World!'
+    sorted_items = sorted(get_items(), key=lambda x: x['status'], reverse=True)
+    return render_template('index.html', items=sorted_items)
+
+@app.route('/add', methods=(['POST']))
+def add():
+    title = request.form.get('title')
+    add_item(title)
+    return redirect(url_for('index'))
+
+@app.route('/done', methods=(['POST']))
+def done():
+    done_items = request.form.getlist('action')
+    for item in done_items[0:-1]:
```

Comment:
This smells like something going wrong - why are we discarding the final element? See my comment below for how we might change it so that we can drop this subtlety and just use:
```suggestion
    for item in done_items:
```